Packs/AzureActiveDirectory/Integrations/AzureADIdentityProtection/AzureADIdentityProtection.py

```diff
@@ -311,6 +311,60 @@ def build_filter(last_fetch, params):
     return query_filter
 
 
+# TODO: move to CommonServerPython?
+from datetime import datetime, timedelta
+from typing import Callable, Dict
+def fetch_last_run(fetch_func: Callable,
+                   params,
+                   time_format_func: Callable = None,
+                   integration_name=''):
+    '''
+      An abstraction for the lastRun fetch flow.
+
+      Args:
+        fetch_func (Callable): User supplied function to fetch incidents,
+          uses last time as a parameter and returns the new time.
+        params (Dict[str, str]): Demisto configuration object
+        time_format_func (Callable): A user defined function to convert date
+                            time into the intergration specified date format.
+        integration_name (str): Integration name as prefix for debug logs.
+      Returns:
+        None
+    '''
+
+    log_prefix = f'[{integration_name}]' if integration_name else '[Debug]'
+
+    last_run = demisto.getLastRun()
+    demisto.debug(f'{log_prefix} last run: {last_run}')
+
+    # If this is the first time running, set to the integration configuration
+    initial_offset_string = f"{params.get('first_fetch') or '1 days'} ago"
+    initial_offset = dateparser.parse(date_string=initial_offset_string)
+                                      #date_formats=[DATE_FORMAT])
+
+    if not last_run.get('last_time_run', None):
+        demisto.debug(f'{log_prefix} first run, setting to initial offset: '
+                      f'{initial_offset_string}')
+        last_run['last_time_run'] = initial_offset
+
+    # If the user supplied a time conversion function, use it.
+    if time_format_func:
```

Comment:
Rename this hook method to something like `fetch_time_formatter`, or `fetch_time_format_hook`, or similar. It's a very generic name as it is.